editor/modes/visual-editor/block.js

```diff
@@ -251,6 +298,18 @@ class VisualEditorBlock extends Component {
 	onKeyDown( event ) {
 		const { keyCode, target } = event;
 
+		this.handleToolbarTabCycle( event );
+		this.handleToolbarArrowCycle( event );
+
+		if ( keyCode === F10 && ! ( event.altKey || event.ctrlKey || event.shiftKey || event.metaKey ) ) {
```

Comment:
I like how you delegated specific logic to helper functions for `handleToolbarTabCycle` and `handleToolbarArrowCycle`. Any reason we shouldn't do the same for the behavior here?